httpd/handler_test.go

```diff
@@ -1118,16 +1118,54 @@ func TestHandler_DropSeries(t *testing.T) {
 func TestHandler_serveWriteSeries(t *testing.T) {
 	c := test.NewMessagingClient()
 	defer c.Close()
-	srvr := OpenAuthenticatedServer(c)
+	srvr := OpenAuthlessServer(c)
 	srvr.CreateDatabase("foo")
-	srvr.CreateRetentionPolicy("foo", influxdb.NewRetentionPolicy("bar"))
 	s := NewHTTPServer(srvr)
 	defer s.Close()
 
-	status, _ := MustHTTP("POST", s.URL+`/write`, nil, nil, `{"database" : "foo", "retentionPolicy" : "bar", "points": [{"name": "cpu", "tags": {"host": "server01"},"timestamp": "2009-11-10T23:00:00Z","fields": {"value": 100}}]}`)
+	status, _ := MustHTTP("POST", s.URL+`/write`, nil, nil, `{"database" : "foo", "retentionPolicy" : "default", "points": [{"name": "cpu", "tags": {"host": "server01"},"timestamp": "2009-11-10T23:00:00Z","fields": {"value": 100}}]}`)
 
 	if status != http.StatusOK {
-		t.Fatalf("unexpected status: %d", status)
+		t.Fatalf("unexpected status for post: %d", status)
+	}
+	query := map[string]string{"db": "foo", "q": "select * from cpu"}
+	status, body := MustHTTP("GET", s.URL+`/query`, query, nil, "")
+	if status != http.StatusOK {
+		t.Fatalf("unexpected status for get: %d", status)
+	}
+	if !strings.Contains(body, `"name":"cpu"`) {
+		t.Fatalf("Write doesn't match query results. Response body is %s.", body)
+	}
+}
+
+func TestHandler_serveDump(t *testing.T) {
+	c := test.NewMessagingClient()
+	defer c.Close()
+	srvr := OpenAuthlessServer(c)
+	srvr.CreateDatabase("foo")
+	s := NewHTTPServer(srvr)
+	defer s.Close()
+
+	status, _ := MustHTTP("POST", s.URL+`/write`, nil, nil, `{"database" : "foo", "retentionPolicy" : "default", "points": [{"name": "cpu", "tags": {"host": "server01"},"timestamp": "2009-11-10T23:00:00Z","fields": {"value": 100}}]}`)
+
+	if status != http.StatusOK {
+		t.Fatalf("unexpected status for post: %d", status)
+
+	}
+	query := map[string]string{"db": "foo", "q": "select * from cpu"}
+	status, body := MustHTTP("GET", s.URL+`/query`, query, nil, "")
+	if status != http.StatusOK {
+		t.Fatalf("unexpected status for get: %d", status)
+	}
+
+	time.Sleep(500 * time.Millisecond) // Shouldn't committed data be readable?
```

Comment:
Writing data is not committed data.  When you do a write, that means that we accepted the message, and are sending it across brokers to write it and mark as committed.  The better practice than sleep is to query and wait.

We do this in other tests here: https://github.com/influxdb/influxdb/blob/master/cmd/influxd/server_integration_test.go#L210

I wouldn't change it for now, but in general, this can lead to a racy test, as well as it artificially inflates our test suite, which adds up over time.